src/platform_impl/linux/x11/mod.rs

```diff
@@ -374,7 +361,12 @@ impl<T: 'static> EventLoop<T> {
 
             // Empty the user event buffer
             {
-                while let Ok(event) = this.user_receiver.try_recv() {
+                while let Some(event) = get_xtarget(&this.target)
+                    .state
+                    .borrow_mut()
```

Comment:
I really don't like those `borrow_mut`, they should generally be `get_mut()`, though, due to flowed design of x11 backend it's sort of that way due to `Rc`(which could easily be removed from what I remember).

Reply:
Okay, I've moved the `state` out of the window target and into the event loop. This should be fine for now.